src/golang/cmd/executor/executor/workflow.go

```diff
@@ -63,6 +64,14 @@ func NewWorkflowExecutor(spec *job.WorkflowSpec, base *BaseExecutor) (*WorkflowE
 }
 
 func (ex *WorkflowExecutor) Run(ctx context.Context) error {
+	// First, ensure that workflow execution is not paused
+	lock := filelock.New(engine.ExecutionLock)
+	// The following will block until the RLock can be acquired
+	if err := lock.RLock(); err != nil {
+		return err
+	}
+	defer lock.RUnlock()
```

Comment:
This is safe even if say the process crashes before it reaches the `defer` statement. This is because the underlying filelock is automatically released when the process exits.